packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -47,6 +48,9 @@ import {
   scheduleSyncCallback,
 } from './SchedulerWithReactIntegration';
 
+// The scheduler is imported here *only* to detect whether it's been mocked
+import * as Scheduler from 'scheduler';
```

Comment:
I could have also exported `isSchedulerMocked` from `SchedulerWithReactIntegration.js` if we want to prevent this import here.